File: BackEnd/ELearning.API/Entites/InstructorApplication.cs
using ELearning.API.Constants;
using System.ComponentModel.DataAnnotations.Schema;

namespace ELearning.API.Entites
{
    public class InstructorApplication
    {
        public int Id { get; set; }

        public string FullName { get; set; }

        public string Email { get; set; }

        public string Phone { get; set; }

        public string Specialization { get; set; }

        public string CvUrl { get; set; }

        public ApplicationStatus Status { get; set; }  

        public DateTime AppliedAt { get; set; } = DateTime.Now;

        public DateTime? ReviewedAt { get; set; }
        public string? Notes { get; set; }
        [ForeignKey("User")]
        public string? UserId { get; set; }
        public User? User { get; set; }  

        [ForeignKey("ReviewedByUser")]
        public string? ReviewedBy { get; set; }
        public User ReviewedByUser { get; set; }
    }
}

File: FrontEnd/src/app/Shared/Models/InstructorApplication.ts
export interface InstructorApplication {
  fullName: string;
  email: string;
  phone: string;
  specialization: string;
  cvUrl: string;
  notes:string;
}

export interface instructorApplicationResponse{
  id: number;
  fullName: string;
  email: string;
  phone: string;
  specialization: string;
  cvUrl: string;
  status: ApplicationStatus;
  appliedAt: string; 
}

export enum ApplicationStatus {
  Pending = 0,
  Approved = 1,
  Rejected = 2
}

export interface ReviewDto{
  applicationId:number,
  isApproved:boolean,
  reviewerAdminEmail:string,
  Notes?:string
}